mpdev/internal/apply/deployment_manager.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apply
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+)
+
+// DeploymentManagerAutogenTemplate generates a deployment manager template
+// given an autogen.yaml file.
+type DeploymentManagerAutogenTemplate struct {
+	ResourceShared
+	AutogenFile string
+	PartnerID   string `json:"partnerId"`
+	SolutionID  string `json:"solutionId"`
+
+	outDir string
+}
+
+// Apply generates a deployment manager template from an autogen file.
+func (dm *DeploymentManagerAutogenTemplate) Apply() error {
+	dir, err := ioutil.TempDir("", "autogen")
+	if err != nil {
+		return err
+	}
+	dm.outDir = dir
+
+	fmt.Printf("Generating deployment manager template from autogen\n")
+	err = dm.convertAutogenSchema()
+	if err != nil {
+		return err
+	}
+
+	autogenImg := "gcr.io/cloud-marketplace-tools/dm/autogen"
+	args := []string{"--input_type", "YAML", "--single_input", "/autogen/autogen.yaml",
+		"--output_type", "PACKAGE", "--output", "/autogen"}
+
+	cp := &ContainerProcess{
+		containerImage: autogenImg,
+		processArgs:    args,
+		mounts:         []string{fmt.Sprintf("type=bind,src=%s,dst=/autogen", dm.outDir)},
```

Comment:
`ContainerProcess` should provide some typed configurations for these mounts, instead of asking for a docker specific string.